File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.js
function solution(num_list) {
    // 홀수만 문자열로 결합
    let oddStr = num_list.filter(num => num % 2 === 1).join('');

    // 짝수만 문자열로 결합
    let evenStr = num_list.filter(num => num % 2 === 0).join('');

    // 문자열을 정수로 변환 후 합산
    return parseInt(oddStr) + parseInt(evenStr);
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.py
def solution(num_list):
    str1 = ''.join(str(num_list[i]) for i in range(len(num_list)) if num_list[i] % 2 == 1)
    str2 = ''.join(str(num_list[i]) for i in range(len(num_list)) if num_list[i] % 2 == 0)
    
    return int(str1) + int(str2)